modules/programs/git.nix

```
@@ -201,6 +201,22 @@ in {
         '';
       };
 
+      hooks = mkOption {
+        type = types.listOf (types.submodule {
+          options = {
+            name = mkOption { type = types.str; };
+            text = mkOption { type = types.str; };
+          };
+        });
```

Would

```suggestion
        type = types.attrsOf types.lines;
```

work? Looking at the `symlinkJoin` call, the name needs to be unique anyway?

Also, presumably one would like to be able to merge the hook content, hence `types.lines`.

---

Good catch. This is definitely nicer.